src/sign.rs

```diff
@@ -1144,4 +1144,139 @@ mod tests {
             }
         }
     }
+
+    #[cfg(feature = "key_tpm")]
+    mod tpm {
+        use super::TEXT;
+        use crate::{crypto::tpm::TpmKey, sign::*};
+
+        use tss_esapi::{
+            attributes::SessionAttributesBuilder,
+            constants::SessionType,
+            interface_types::{
+                algorithm::HashingAlgorithm, ecc::EccCurve, resource_handles::Hierarchy,
+            },
+            structures::SymmetricDefinition,
+            utils::{create_unrestricted_signing_ecc_public, AsymSchemeUnion},
+            Context, Tcti,
+        };
+
+        #[test]
+        fn cose_sign_tpm() {
+            let mut tpm_context = unsafe {
+                Context::new(Tcti::from_environment_variable().expect("Failed to get TCTI"))
+                    .expect("Failed to create context")
+            };
+            let tpm_session = tpm_context
+                .start_auth_session(
+                    None,
+                    None,
+                    None,
+                    SessionType::Hmac,
+                    SymmetricDefinition::AES_128_CFB,
+                    HashingAlgorithm::Sha256,
+                )
+                .expect("Error creating TPM session")
+                .expect("Expected AuthSession");
+            let (session_attrs, session_attrs_mask) = SessionAttributesBuilder::new()
+                .with_decrypt(true)
+                .with_encrypt(true)
+                .build();
+            tpm_context
+                .tr_sess_set_attributes(tpm_session, session_attrs, session_attrs_mask)
+                .expect("Error setting session attributes");
+            tpm_context.set_sessions((Some(tpm_session), None, None));
+            let prim_key = tpm_context
```

Comment:
Given that this creates a new primary key, could running this test a few times end up consuming all the NVRAM of the TPM? I think the test should remove the key after it's done with it.

Reply:
No, nothing is stored in the NVRAM.
TPM 2.0's do not store any keys in the NVRAM, since they're derived from the input data every time you call `create_(primary_)key`.
The one thing that might get full is the handle space, but tss-esapi should automatically clear those when the context is dropped.